draft-law-moq-cat4moqt.md

```diff
@@ -333,6 +333,314 @@ claims, the token is not well-formed.
+
+Window Setting (key 0): Defines acceptable time window for DPoP proof validity
+in seconds
+
+- Type: Unsigned integer
+- Usage: Relays MUST use this value to validate DPoP proof timestamps
+- Example: 300 (5-minute acceptance window)
+
+JTI Setting (key 1): Controls processing of "jti" claim in DPoP proofs for
+replay protection
+
+- Value 0: Ignore "jti" claims (no replay protection)
+- Value 1: Honor "jti" claims when available for replay protection
+- Default: 0 if not specified
+
+Critical Setting (key -1): Array of setting keys that MUST be understood
+
+- Type: Array of integers
+- Usage: Recipients MUST reject "catdpop" claim if they don't understand
+  listed critical settings
+- Restriction: MUST NOT contain -1, 0, or 1 (these are always required to
+  be understood)
+
+### MOQT Action to HTTP Method Mapping
+
+For DPoP "htm" claim construction, MOQT actions map to HTTP methods as follows:
+
+|----------------------|-----|
+| MOQT Action          | HTM |
+|----------------------|-----|
+| CLIENT_SETUP         | POST|
+| SERVER_SETUP         | POST|
+| ANNOUNCE             | PUT |
+| SUBSCRIBE_NAMESPACE  | GET |
+| SUBSCRIBE            | GET |
+| PUBLISH              | POST|
+| FETCH                | GET |
+|----------------------|-----|
+
+### Alternate Proposal: DPoP Extension with Generic Command Type Framework
+
+This section proposes an alternate approach where DPoP is extended to
+support a generic command type framework, allowing for more flexible and
+protocol-specific command representations. Instead of constraining all
+actions to HTTP method semantics, this framework introduces a "cmd" (command)
+claim that can accommodate different command type systems.
+
+
+The DPoP proof JWT is extended with an optional "cmd" (command) claim that
+works alongside or replaces the traditional "htm" (HTTP method) claim, as
+shown in the example below:
+
+~~~~~~~~~~~~~~~
+{
+  "typ": "dpop+jwt",
+  "alg": "ES256",
+  "jwk": { ... }
+}
+.
+{
+  "jti": "unique-request-id",
+  "iat": 1705123456,
+  "htu": "moqt://relay.example.com/sports/live-feed",
+  "cmd": {
+    "type": "moqt",
+    "value": "ANNOUNCE"
+  }
+}
+~~~~~~~~~~~~~~~
+
+
+This proposal introduces native MOQT command semantics that better
+represent the actual protocol operations:
+
+|----------------------|------------------------------------|
+| MOQT Action          | Generic Command                    |
+|----------------------|------------------------------------|
+| CLIENT_SETUP         | {"type":"moqt","value":"SETUP"}    |
+| SERVER_SETUP         | {"type":"moqt","value":"SETUP"}    |
+| ANNOUNCE             | {"type":"moqt","value":"ANNOUNCE"} |
+| SUBSCRIBE_NAMESPACE  | {"type":"moqt","value":"SUB_NS"}   |
+| SUBSCRIBE            | {"type":"moqt","value":"SUBSCRIBE"}|
+| PUBLISH              | {"type":"moqt","value":"PUBLISH"}  |
+| FETCH                | {"type":"moqt","value":"FETCH"}    |
+|----------------------|------------------------------------|
+
```

Comment:
I have a draft write up in progress on these lines . I plan to post the details soon